workers/unity/Assets/Fps/Scripts/GameLogic/StateMachine/Session/PrepareDeploymentsListState.cs

```diff
@@ -44,30 +41,27 @@ private void BrowseDeployments()
 
         private void Connect()
         {
-            Owner.SetState(new ConnectState(deployment: null, Manager, Owner));
+            Owner.SetState(new SessionConnectState(deployment: null, Manager, Owner));
         }
 
         private bool PrepareDeploymentsList()
         {
-            if (Owner.Blackboard.LoginTokens == null)
+            if (Blackboard.LoginTokens == null)
             {
                 Owner.SetState(Owner.StartState);
             }
 
-            var hasAvailableDeployment = false;
             var deploymentData = new List<DeploymentData>();
-            foreach (var loginToken in Owner.Blackboard.LoginTokens)
+            foreach (var loginToken in Blackboard.LoginTokens)
```

Comment:
You know much much I like LINQ... 

```csharp
var deploymentData = Blackboard.LoginTokens
	.Select(token => (DeploymentData.TryFromTags(token, out var data), data))
	.Where(pair => pair.First)
	.Select(pair => pair.Second)
	.ToList()
```

Even better if we have a `filter_map` operation: 

```csharp
var deploymentData = Blackboard.LoginTokens
	.FilterMap(token => (DeploymentData.TryFromTags(token, out var data), data))
	.ToList()
```